clients/task/task.go

```diff
@@ -52,9 +54,217 @@ func (c Client) Create(ctx context.Context, params *CreateParams) error {
+	if params.TaskID != "" {
+		task, err := c.GetTasksID(ctx, params.TaskID).Execute()
+		if err != nil {
+			return err
+		}
+		tasks = append(tasks, task)
+	} else {
+		// filter on all tasks
+		taskGet := c.GetTasks(ctx).Org(org).Limit(int32(params.Limit))
+		if params.UserID != "" {
+			taskGet = taskGet.User(params.UserID)
+		}
+		tasksResult, err := taskGet.Execute()
+		if err != nil {
+			return err
+		}
+		tasks = *tasksResult.Tasks
+	}
+	return c.printTasks(taskPrintOpts{
+		tasks: tasks,
+	})
+}
+
+func (c Client) appendRuns(ctx context.Context, prev []api.Run, taskID string, filter RunFilter) ([]api.Run, error) {
+	if filter.Limit < 1 {
+		return nil, fmt.Errorf("must specify a positive run limit, not %d", filter.Limit)
+	}
+	getRuns := c.GetTasksIDRuns(ctx, taskID).Limit(int32(filter.Limit))
+	if filter.After != "" {
+		afterTime, err := time.Parse(time.RFC3339, filter.After)
+		if err != nil {
+			return nil, err
+		}
+		getRuns = getRuns.AfterTime(afterTime)
+	}
+	if filter.Before != "" {
+		beforeTime, err := time.Parse(time.RFC3339, filter.Before)
+		if err != nil {
+			return nil, err
+		}
+		getRuns = getRuns.BeforeTime(beforeTime)
+	}
+	runs, err := getRuns.Execute()
+	if err != nil {
+		return nil, err
+	}
+	for _, run := range *runs.Runs {
+		if filter.Status == "" {
+			prev = append(prev, run)
+		} else if run.Status != nil && *run.Status == filter.Status {
+			prev = append(prev, run)
+		}
+	}
+	return prev, nil
+}
+
+type RunFilter struct {
+	After  string
+	Before string
+	Limit  int
+	Status string
+}
+
+type RetryFailedParams struct {
+	clients.OrgParams
+	TaskID    string
+	DryRun    bool
+	TaskLimit int
+	RunFilter RunFilter
+}
+
+func (c Client) retryRun(ctx context.Context, run api.Run, dryRun bool) error {
+	// Note that this output does not respect json flag, in line with original influx cli
+	// The server should fill in the empty id's so this shouldn't happen
+	if run.Id == nil {
+		_ = c.StdIO.Error("skipping empty run id from influxdb")
+		return nil
+	}
+	if run.TaskID == nil {
+		_ = c.StdIO.Error("skipping empty task id from influxdb")
+		return nil
+	}
+	if dryRun {
+		_, _ = fmt.Fprintf(c.StdIO, "Would retry for %s run for Task %s.\n", *run.Id, *run.TaskID)
+	} else {
+		newRun, err := c.PostTasksIDRunsIDRetry(ctx, *run.TaskID, *run.Id).Execute()
+		if err != nil {
+			return err
+		}
+		_, _ = fmt.Fprintf(c.StdIO, "Retry for task %s's run %s queued as run %s.\n", *run.TaskID, *run.Id, *newRun.Id)
+	}
+	return nil
+}
+
+func (c Client) RetryFailed(ctx context.Context, params *RetryFailedParams) error {
+	if params.TaskLimit < 1 {
+		return fmt.Errorf("must specify a positive task limit, not %d", params.TaskLimit)
+	}
+	var taskIds []string
+	if params.TaskID != "" {
+		taskIds = []string{params.TaskID}
+	} else {
+		org, err := c.getOrg(&params.OrgParams)
+		if err != nil {
+			return err
+		}
+		tasks, err := c.GetTasks(ctx).Limit(int32(params.TaskLimit)).Org(org).Execute()
```

Comment:
Same comments as above:
* `int32` and `Limit`
* Will `Org(org)` work if `org` is an ID?